lexicon/community/bookmark.json

```diff
@@ -0,0 +1,19 @@
+{
+    "lexicon": 1,
+    "id": "community.lexicon.bookmark",
+    "defs": {
+      "main": {
+        "type": "record",
+        "description": "Record bookmarking a link to come back to later.",
+        "key": "tid",
+        "record": {
+          "type": "object",
+          "required": ["subject", "createdAt"],
+          "properties": {
+            "subject": { "type": "ref", "ref": "com.atproto.repo.strongRef" },
```

Comment:
What do you think about making this a string with the format `uri`[^lexiconschema]. It'd allow anything that passes as a URI to qualify, which means that people could store any bookmarks in their repository. The `uri` format qualifies as anything conforming to [RFC-3986](https://datatracker.ietf.org/doc/html/rfc3986#section-1.1) for validation purposes.

```json
{
    "$type": "community.lexicon.bookmark",
    "createdAt": "2024-12-06T11:00:00.000Z",
    "subject": "at://did:plc:2uwoih2htodskvgocarwv5eq/app.bsky.feed.post/3lc6kjp5vuc2o"
}
```

[^lexiconschema]: https://atproto.com/specs/lexicon#string-formats

Reply:
I'm personally pro-strongref. If you decide you don't need the CID, you can just ignore it, and I think it's better to "have and not need" vs "need and not have", as it were.

If you want to bookmark some non at:// resource, you could make a post that contains it (or any other record type) and bookmark *that* instead.

Flexibility is nice but I think it's important to justify it vs whatever implementation complexity it might entail.

Reply:
I think both points are valid but leaning more towards strongref as well. Thinking through realistic bookmarking scenarios, bookmarks are used to save things like posts, media, etc. If you want to bookmark something like a profile, you would presumably use a different lexicon like [follow](https://github.com/bluesky-social/atproto/blob/main/lexicons/app/bsky/graph/follow.json).

Reply:
> If you want to bookmark some non at:// resource, you could make a post that contains it (or any other record type) and bookmark that instead.

I really don't think it's practical to say that if anyone wants to bookmark a website, for example, then they have to create a post with an embed (or link?) and then create an additional record to represent that it's bookmarked.

By keeping it as a URI, which all AT-URIs are compatible with, this one common record type can be used beyond just bookmarking only records.

```json
{
    "$type": "community.lexicon.bookmark",
    "createdAt": "2024-12-06T11:00:00.000Z",
    "subject": "https://example.com/this-great-non-atproto-thing"
}
```

Reply:
I'm not opposed to making it more open because I can see use cases for wanting to bookmark non AT links. I think it would push some complexity onto the app views for handling AT:// and non AT URLs which could or could not be linking to content. But I guess that is fine and aligns with the overall architecture of the protocol where app views can filter out whatever types they choose (bsky can choose to show only AT URLS)

Reply:
That's the thing about community lexicon (at least IMO) is that we want these to be more universal and not too specific that they aren't as useful. While pointing to a specific record is universal on atproto for things that end up being records to say that if you want to "bookmark" a link outside of atproto you must make a post and bookmark that. Also there are reason you might want to bookmark a profile and not follow. Also follow so far is app specific. These lexicon are meant to go well beyond Bluesky as an app.